pkg/fsp/header_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017-2018 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fsp
+
+import (
+	"bytes"
+	"io/ioutil"
+	"testing"
+)
+
+const (
+	// From https://github.com/IntelFsp/FSP/blob/master/ApolloLakeFspBinPkg/FspBin/Fsp.fd
+	// under the FSP license. See README.md under `test_blobs`.
+	FSPTestFile = "test_blobs/ApolloLakeFspBinPkg/Fsp.fd"
+)
+
+func TestFromBytesHeaderV3(t *testing.T) {
+	buf, err := ioutil.ReadFile(FSPTestFile)
+	if err != nil {
+		t.Errorf("Error opening test file %s: %v", FSPTestFile, err)
+	}
+	// in the ApolloLake FSP, the FSP header starts at byte 148. This can be
+	// extracted by parsing it as uefi.FirmwareVolume, but it's not relevant for
+	// this test, so let's go with raw offsets.
+	hdr, err := FromBytes(buf[148 : 148+HeaderV3Length])
```

Comment:
One possibility to avoid this is to dump the exact binary we care about, and get rid of the wrapping headers.

Reply:
I prefer to use the original FSP files from Intel, because at some point I want to introduce some integration tests. I agree with you that for unit tests the raw sequence of bytes would be more appropriate, and I can put that in the unit test file

Reply:
maybe I can test the CLI with the whole FSP file as an integration test, and the pkg with the golden header as a unit test. Would that work?

Reply:
done that way